changelog/0032-upgrade-elasticsearch.yml

```diff
@@ -0,0 +1,84 @@
+---
+title: ES upgrade from 1.7.6 to 2.4.6
+key: upgrade-elasticsearch
+date: 2020-02-05
+optional_per_env: no
+min_commcare_version: 19be50699bf78b25a8da6e3e13eb5bc6d13245e8
+context: |
+  This change upgrade Elasticsearch from 1.7.6 to 2.4.6 version.
+
+details: |
+  As part of our ongoing effort to keep CommCare HQ up to date with the latest tools and
+  libraries we have updated Elasticsearch from 1.7.6 to 2.4.6 version.
+
+update_steps: |
+
+  1. Stop the site
+  ```bash
+  commcare-cloud <env> downtime start
+  ```
+    
+  2. Add the following parameters to your environment's `public.yml`:
+  ``` 
+  elasticsearch_version: 2.4.6
+  elasticsearch_download_sha256: 5f7e4bb792917bb7ffc2a5f612dfec87416d54563f795d6a70637befef4cfc6f.
+  ELASTICSEARCH_MAJOR_VERSION: 2
+  ```
+    
+  3. Update the local settings
+  ```bash
+  commcare-cloud <env> update-config
+  ```  
+  4. Disable the cluster routing
+  ```bash
+  curl -XPUT <ES-IP>:9200/_cluster/settings -d '{\"persistent\" : {\"cluster.routing.allocation.enable\" : \"none\" }}'
```

Comment:
nit: I don't think you need to escape the `"`s here with a `\` since they're wrapped with `'`. You also might be required to have the content type application/json header, though I'm not 100% sure. I'd suggest you go to https://www.elastic.co/guide/en/elasticsearch/reference/2.4/restart-upgrade.html and click on "Copy as cURL" for each of these curl commands, and then replace "localsettings" with `$(cchq ${ENV} lookup elasticsearch:0)`

Reply:
Throughout instead of <ES-IP>, you can use $(cchq ${ENV} lookup elasticsearch:0) which will actually calculate that. Since there are so many commands to run, I might also tell them to set their ENV environment variable, and then use ${ENV} throughout these commands

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/2.4/restart-upgrade.html#_step_1_disable_shard_allocation_2 also suggests performing a `POST /_flush/synced` before shutting down.